native-assets/fractional-NFT/project/contracts/f-NFT-contract/src/main.sw

```diff
@@ -0,0 +1,475 @@
+contract;
+
+mod errors;
+
+use errors::{DepositError, SubIdError, WithdrawError,};
+use src6::{Deposit, SRC6, Withdraw};
+use src20::SRC20;
+use std::{
+    auth::msg_sender,
+    call_frames::{
+        contract_id,
+        msg_asset_id,
+    },
+    constants::ZERO_B256,
+    context::{
+        msg_amount,
+        this_balance,
+    },
+    hash::{
+        Hash,
+        sha256,
+    },
+    string::String,
+    token::{
+        burn,
+        mint_to,
+        transfer,
+    },
+};
+
+configurable {
+    /// The decimals of fractionalized NFT assets minted by this contract.
+    DECIMALS: u8 = 9u8,
+    /// The name of fractionalized NFT assets minted by this contract.
+    NAME: str[18] = __to_str_array("Fractionalized NFT"),
+    /// The symbol of fractionalized NFT assets minted by this contract.
+    SYMBOL: str[4] = __to_str_array("FNFT"),
+    /// The number of shares per NFT
+    SHARES: u64 = 100_000_000,
+}
+
+storage {
+    /// The total number of unique assets minted by this contract.
+    ///
+    /// # Additional Information
+    ///
+    /// This is the number of fractional NFTs that have ever been deposited.
+    total_assets: u64 = 0,
+    /// The validity of an asset as a share minted by this contract.
+    ///
+    /// # Additional Information
+    ///
+    /// maps(Fractional NFT Share) => valid share
+    vault_asset: StorageMap<AssetId, bool> = StorageMap {},
```

Comment:
This sounds strange without already being familiar with the code.
Perhaps rename to "valid_asset"?

Reply:
Valid asset would imply NFTs but this is only the assets minted by this contract aka the vault assets.